gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java

```diff
@@ -787,7 +787,8 @@ private void staticHandler(Router router) {
     }
 
     private void bodyHandler(Router router) {
-        router.route().handler(BodyHandler.create());
+        router.route()
+                .handler(new io.gravitee.am.gateway.handler.root.resources.handler.ConditionalBodyHandler(environment));
```

Comment:
Suggestion: perhaps you can import the class and then it will be just `new ConditionalBodyHandler`